### PR TITLE
Add the ability to conditionally trigger builds based on changes.

### PR DESCRIPTION
**Pull request by rmmh**
For the cross build, run if a crossplatform or build-like file has
changed.

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -276,6 +276,39 @@ func (c *Client) GetPullRequest(org, repo string, number int) (*PullRequest, err
 	return &pr, nil
 }
 
+// GetPullRequestChanges gets a list of files modified in a pull request.
+func (c *Client) GetPullRequestChanges(pr PullRequest) ([]PullRequestChange, error) {
```
Comment (spxtr) [inline prow/github/client.go]: Add a simple test like for list issue comments.

**Comment by spxtr** [comment]
Overall seems good. We might want to consider setting the status to pass for jobs we're skipping like we do for jobs on old branches.

File: prow/presubmit.yaml [hunk 2]
```diff
@@ -78,6 +78,7 @@ kubernetes/kubernetes:
   context: Jenkins Cross Build
   rerun_command: "@k8s-bot build this"
   trigger: "@k8s-bot (cross )?build this"
+  run_if_changed: '(Makefile|\.sh|_(windows|linux|osx|unknown)(_test)?\.go)$'
```
Comment (luxas) [inline prow/presubmit.yaml]: I think it should be unsupported instead of unknown

Comment (rmmh) [inline prow/presubmit.yaml]: Done.

**Comment by rmmh** [comment]
@spxtr PTAL

File: prow/jobs/jobs_test.go [hunk 3]
```diff
@@ -215,6 +215,34 @@ func TestCommentBodyMatches(t *testing.T) {
...
+	var testcases = []struct {
+		changes  []string
+		expected bool
+	}{
+		{[]string{"some random file"}, false},
+		{[]string{"./pkg/util/rlimit/rlimit_linux.go"}, true},
+		{[]string{"./pkg/util/rlimit/rlimit_unknown_test.go"}, true},
+		{[]string{"build.sh"}, true},
+		{[]string{"build.shoo"}, false},
+		{[]string{"Makefile"}, true},
```
Comment (spxtr) [inline prow/jobs/jobs_test.go]: Include a couple cases with multiple files.

File: prow/presubmit.yaml [hunk 4]
```diff
@@ -80,6 +80,9 @@ kubernetes/kubernetes:
   skip_report: true
   rerun_command: "@k8s-bot cross build this"
   trigger: "@k8s-bot (cross[ -])?(build|test) this"
+  #  rerun_command: "@k8s-bot build this"
```
Comment (spxtr) [inline prow/presubmit.yaml]: ?

**Comment by rmmh** [comment]
I don't think setting a passing status for something that doesn't actually run is a good idea.

I'd rather modify the munger to have the concept of 'ignore if missing, block on if present'-- so if there's a failing cross build it won't put it in the queue, but if there's no cross build it will.

**Comment by spxtr** [comment]
Then we have to do away with required status checks on github. I'm okay with that, approach, but I think it's more work than just setting the status to pass.

**Comment by rmmh** [comment]
The required status checks already don't match reality and are ignored by the bot.

**Comment by spxtr** [comment]
Right, but we do take some effort to make them match. Here's a strange case that will probably not happen but might:

1. Make a PR that touches `something_linux.go` and is totally broken. The cross-build fails.
1. Update your PR so that it no longer touches that file.
1. Uh oh! You still have an X!

**Pull request merged**